pyomo/core/tests/examples/test_pyomo.py

```diff
@@ -106,7 +106,34 @@ def compare_json(self, file1, file2):
                                              allow_second_superset=True)
 
     def compare_files(self, file1, file2):
-        self.assertTrue(file1, file2)
+        try:
+            self.assertTrue(cmp(file1, file2),
+                        msg="Files %s and %s differ" % (file1, file2))
+        except:
+            with open(file1, 'r') as f1, open(file2, 'r') as f2:
+                f1_contents = f1.read().strip().split('\n')
+                f2_contents = f2.read().strip().split('\n')
+                f1_filtered = []
+                f2_filtered = []
+                for item1, item2 in zip(f1_contents, f2_contents):
```

Comment:
This should probably use `itertools.zip_longest` so that differing length files cause errors.